app/observer.tsx

```diff
@@ -0,0 +1,62 @@
+export type Notita = {
+  id: string;
+  text: string;
+  data: string;
+  ora: string;
+};
+
+type Subscriber = (notite: Notita[]) => void;
+
+const ListaNotite: {
+  notite: Notita[];
+  subscribers: Subscriber[];
+
+  subscribe: (callback: Subscriber) => void;
+  unsubscribe: (callback: Subscriber) => void;
+
+  setNotite: (notite: Notita[]) => void;
+  addNotita: (notita: Notita) => void;
+  clearNotite: () => void;
+  getNotite: () => Notita[];
+  notify: () => void;
+} = {
+  notite: [],
+  subscribers: [],
+
+  //pot scrie this. in loc de Lista subscribers facand referinta la obiectul curent
+  notify() {
+    // Trimite o copie a listei tuturor subscriberilor
+    this.subscribers.forEach((callback) => callback([...this.notite])); //callback apel la functia observer
+  },
+
+  subscribe(callback) {
```

Comment:
Do not use subscribe(callback), use subscribe: (callback) => {
}

The difference between functions expressions and functions declarations, arrow functions (aligns better with TS & React Functional Components)